src/schema/posts.ts

```diff
@@ -2092,6 +2122,76 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
 
       return { _: true };
     },
+    moderateSourcePosts: async (
+      _,
+      {
+        sourceId,
+        postIds,
+        status,
+        rejectionReason = null,
+        moderatorMessage = null,
+      }: {
+        postIds: string[];
+      } & Pick<
+        SourcePostModeration,
+        'sourceId' | 'status' | 'rejectionReason' | 'moderatorMessage'
+      >,
+      ctx: AuthContext,
+    ) => {
+      const ids = Array.from(new Set(postIds));
+      const haveDuplicates = ids.length !== postIds?.length;
+      const tooManyIds = ids.length > POST_MODERATION_PAGE_SIZE;
+      if (!postIds.length || haveDuplicates || tooManyIds) {
+        throw new ValidationError('Invalid array of post IDs provided');
+      }
+
+      if (
+        ![
+          SourcePostModerationStatus.Approved,
+          SourcePostModerationStatus.Rejected,
+        ].includes(status)
+      ) {
+        throw new ValidationError('Invalid status provided');
+      }
+
+      await ensureSourcePermissions(
+        ctx,
+        sourceId,
+        SourcePermissions.MemberRemove,
+      );
+
+      const pendingPosts = await ctx.con
+        .getRepository(SourcePostModeration)
+        .find({
+          where: {
```

Comment:
Probably add `select: id` here to avoid requesting the whole object.

Reply:
Updated ✔️